library/kani_core/src/mem.rs

```diff
@@ -149,7 +149,14 @@ macro_rules! kani_mem {
         )]
         #[allow(clippy::not_unsafe_ptr_arg_deref)]
         pub fn same_allocation<T: ?Sized>(ptr1: *const T, ptr2: *const T) -> bool {
-            cbmc::same_allocation(ptr1, ptr2)
+            same_allocation_internal(ptr1, ptr2)
+        }
+
+        #[allow(clippy::not_unsafe_ptr_arg_deref)]
+        pub(super) fn same_allocation_internal<T: ?Sized>(ptr1: *const T, ptr2: *const T) -> bool {
```

Comment:
Nit: I think it would be cleaner to delete this function and just do this:

```rust
#[allow(clippy::not_unsafe_ptr_arg_deref)]
pub fn same_allocation<T: ?Sized>(ptr1: *const T, ptr2: *const T) -> bool {
  let addr1 = ptr1 as *const ();
  let addr2 = ptr2 as *const ();
  cbmc::same_allocation(addr1, addr2)
}
```
and call `same_allocation` inside of our offset model. This isn't a huge deal--it just seems simpler to keep one function, unless I'm missing something here.

Reply:
The reason why I created a second one is because the first one is tagged with `#[unstable]` attribute. Using that one would require us to either remove the attribute or users would have to pass `-Z mem-predicates`.

Reply:
Ah okay, works for me then.